anchor-service/src/anchor_batch.rs

```diff
@@ -76,28 +77,42 @@ impl AnchorService {
         pin_mut!(shutdown_signal);
 
         info!("anchor service started");
+        let period =
```

Comment:
nitpick, but could we pull all this logic for building the interval into a helper function to improve readability?